task-maker-format/src/ioi/sanity_checks/att.rs

```diff
@@ -400,3 +402,42 @@ impl SanityCheck<IOITask> for AttWithNoCheck {
         Ok(())
     }
 }
+
+/// Check that the template and grader in att compile together
+#[derive(Debug, Default)]
+pub struct AttTemplatesShouldCompile;
+
+impl SanityCheck<IOITask> for AttTemplatesShouldCompile {
+    fn name(&self) -> &'static str {
+        "AttTemplatesShouldCompile"
+    }
+
+    fn pre_hook(&mut self, task: &IOITask, eval: &mut EvaluationData) -> Result<(), Error> {
+        for grader in task.grader_map.all_paths() {
+            let ext = grader
+                .extension()
+                .ok_or_else(|| anyhow!("Grader has no extension"))?
+                .to_string_lossy();
+            let att_name = format!("att/{}.{}", task.name, ext);
+            let template = task.path.join(&att_name);
+
+            let grader_name = grader
+                .file_name()
+                .ok_or_else(|| anyhow!("Grader has no file name"))?
+                .to_string_lossy();
+            let att_grader = format!("att/{}", grader_name);
+
+            let grader_map = GraderMap::new(vec![att_grader]);
+
+            if let Some(source_file) = SourceFile::new(
+                template,
+                &task.path,
+                Some(Arc::new(grader_map)),
+                None::<String>,
+            ) {
+                source_file.prepare(eval)?;
```

Comment:
I've noticed that [`task_maker_format::SourceFile::prepare()`](https://edomora97.github.io/task-maker-rust/task_maker_format/struct.SourceFile.html#method.prepare) does not return the compilation execution, while its inner implementation [`task_maker_lang::SourceFile::prepare()`](https://edomora97.github.io/task-maker-rust/task_maker_lang/struct.SourceFile.html#method.prepare) does. You may want to update the former returning the `Option<ExecutionUuid>` for binding a callback and print a nicer warning message.

Reply:
Also, if you manage to add a test for this it would be awesome!

Reply:
Yep, looking into this just now!